pkgs/games/steam/fhsenv.nix

```diff
@@ -110,7 +110,7 @@ in buildFHSEnv rec {
     gnome2.GConf
     curlWithGnuTls
     nspr
-    nss
+    nss_latest
```

Comment:
Firefox uses nss_latest, while firefox-esr uses nss, and ideally it would work with both, so why doesn't it? I don't think this can be the correct solution.

Reply:
yes, i was thinking the same thing,  this is a recurring issue https://github.com/NixOS/nixpkgs/pull/237123

I havent had a chance to look into it, but I think because

ldd /nix/store/d4qvavkldy88djia02866dmkrv200ir4-firefox-114.0.2/bin/.firefox-wrapped

doesnt list nss, it must be dynamically loaded elsewhere, so LD_LIBRARY_PATH takes precedence  and all FHSEnv env will have issues

Reply:
greetings ! I had one more look

the issue lies as i guessed with ld.so

```
readelf -d libxul.so 

Dynamic section at offset 0x866acb0 contains 90 entries:
  Tag        Type                         Name/Value
 0x000000000000001d (RUNPATH)            Library runpath: [/nix/store/b5jw7x8r4jvn65aqfccz1pyakfj75538-icu4c-72.1/lib:/nix/store/qzs91y7s4rwl7mihvh87h8zs29hzcs2i-libffi-3.4.4/lib:/nix/store/fbiggz755ajhjpv615fgnr64imjghbsh-nspr-4.35/lib:/nix/store/s9zl0rv45yc3ymx61q6n97c3l7196pq4-zlib-1.2.13/lib:/nix/store/a1c77grsbrx4jkg4xkbmaj377alwc916-nss-3.90/lib64:/nix/store/gkh0lkj9wmlkhbkkpf8fp7zjf1521pyl-freetype-2.13.0/lib:/nix/store/gxwsa0w5f8krfgwfk9axn58hpa6pbiy2-fontconfig-2.14.2-lib/lib:/nix/store/zw74yn8wwlycd4cb7iwjls3m39qpsmxi-gtk+3-3.24.38/lib:/nix/store/bx9i3hq5lywpw15maixvk5qhsbdrkfag-at-spi2-core-2.48.3/lib:/nix/store/cgmbz0wglid7v5m0m0a70ksvgyxppsgn-glib-2.76.3/lib:/nix/store/bfb2b8flqm6y60s96s6wk4v5b9j4b595-cairo-1.16.0/lib:/nix/store/y306322qv2dh5qrb5f3hgj9rmdpkzxb4-gdk-pixbuf-2.42.10/lib:/nix/store/ah4s45l5n1mgmcvxph6l7bp4rv3azkdq-pango-1.50.14/lib:/nix/store/9rxm4yi7blwl4pwpklfibznl86gfn7h7-harfbuzz-7.3.0/lib:/nix/store/l6x7jz2qkriwf1ms80cpln0kfnly75y5-libpng-apng-1.6.39/lib:/nix/store/zs8lf0digandvqi5ilqinsabzylazswk-libwebp-1.3.0/lib:/nix/store/i4mwwpgwfihb14nkd99dcqx6s2i8kaxw-libevent-2.1.12/lib:/nix/store/la6sj82y46wpfg7h59chf0wfz2nz4gy1-libvpx-1.13.0/lib:/nix/store/wz1dy3d3064wbrvs7wcf9d7jr4f02073-pixman-0.42.2/lib:/nix/store/204j1yqxw4iin087p53x9jg5pa02yr65-dbus-1.14.8-lib/lib:/nix/store/k8s01b4srjrm7ymdajqgqqiwjs1s29w5-dbus-glib-0.112/lib:/nix/store/bnpy6ypmsi21gym32qj8yb26sbsmf8bq-libXext-1.3.4/lib:/nix/store/nwym4ys464vziikfjwmc9ngamcfaygvm-libX11-1.8.4/lib:/nix/store/99kyqdivq646j0w4smh2bfqd2pkrcrai-libxcb-1.14/lib:/nix/store/df9c7n5hdx7ml0jrvrv7blmkygznpz81-libXcomposite-0.4.5/lib:/nix/store/q2r07izn69ipnhkjdsnwz28p0rzfas0a-libXfixes-6.0.0/lib:/nix/store/h449fmwdwj5dx3qk45shzxpx7hx4iciq-libXcursor-1.2.0/lib:/nix/store/9dvq1hgk5fvrnjq4sd6w9m7m0zha70mf-libXdamage-1.1.5/lib:/nix/store/y4bkflfiprrz1f7zkax1707rwqc7aj0x-libXi-1.8/lib:/nix/store/n32rla0prb7hi1ycrrhqks6i8b9zc35q-libXrandr-1.5.2/lib:/nix/store/w4lwvqgf0v5846ygqfq0zga54fvbrgpv-libXtst-1.2.3/lib:/nix/store/ylwqkikzvl571iql3cj6ghqzis7d5kjp-libXrender-0.9.10/lib:/nix/store/zc7vf2b6qkacgdkqkna6aa7afhnfmr0x-alsa-lib-1.2.9/lib:/nix/store/wpgrc564ys39vbyv0m50qxmq8dvhi7cc-glibc-2.37-8/lib:/nix/store/d4wb4vw0i85xyy7qs82l4s4mr2yigvc9-gcc-12.3.0-libgcc/lib]
 0x0000000000000001 (NEEDED)             Shared library: [libpthread.so.0]
 0x0000000000000001 (NEEDED)             Shared library: [libmozsandbox.so]
 0x0000000000000001 (NEEDED)             Shared library: [liblgpllibs.so]
 0x0000000000000001 (NEEDED)             Shared library: [libmozsqlite3.so]
 0x0000000000000001 (NEEDED)             Shared library: [libmozgtk.so]
 0x0000000000000001 (NEEDED)             Shared library: [libmozwayland.so]
 0x0000000000000001 (NEEDED)             Shared library: [libicui18n.so.72]
 0x0000000000000001 (NEEDED)             Shared library: [libicuuc.so.72]
 0x0000000000000001 (NEEDED)             Shared library: [libicudata.so.72]
 0x0000000000000001 (NEEDED)             Shared library: [libasound.so.2]
 0x0000000000000001 (NEEDED)             Shared library: [librt.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libdl.so.2]
 0x0000000000000001 (NEEDED)             Shared library: [libX11.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libXcomposite.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libXdamage.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libXext.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libXfixes.so.3]
 0x0000000000000001 (NEEDED)             Shared library: [libXrandr.so.2]
 0x0000000000000001 (NEEDED)             Shared library: [libXrender.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libXtst.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [ld-linux-x86-64.so.2]
 0x0000000000000001 (NEEDED)             Shared library: [libffi.so.8]
 0x0000000000000001 (NEEDED)             Shared library: [libplds4.so]
 0x0000000000000001 (NEEDED)             Shared library: [libplc4.so]
 0x0000000000000001 (NEEDED)             Shared library: [libnspr4.so]
 0x0000000000000001 (NEEDED)             Shared library: [libz.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libssl3.so]
 0x0000000000000001 (NEEDED)             Shared library: [libsmime3.so]
 0x0000000000000001 (NEEDED)             Shared library: [libnss3.so]

```

as per 
https://man7.org/linux/man-pages/man8/ld.so.8.html

LD_LIBRARY_PATH takes precedence over RUNPATH

so either nix machinery needs to write full paths for shared librarries or firefox binary needs to set those in RPATH

I am not educated enough to suggest which approach is better